### PR TITLE
Refactor fonts from dict to classes

### PR DESCRIPTION
**Pull request by andersonhc**
This PR is only refactoring the code touching fonts. 
Instead of dictionaries, I created the classes TTFFont and CoreFont

This is a first step for the text shaping functionality

**Checklist**:

- [x] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [ ] A unit test is covering the code added / modified by this PR

- [x] This PR is ready to be merged 

- [ ] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [ ] A mention of the change is present in `CHANGELOG.md`

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/PyFPDF/fpdf2/blob/master/LICENSE).

### CODE REVIEW

File: fpdf/fpdf.py [hunk 1]
```diff
@@ -368,7 +308,7 @@ def __init__(
         self.font_stretching = 100  # current font stretching
         self.char_spacing = 0  # current character spacing
         self.underline = False  # underlining flag
-        self.current_font = {}  # current font
+        self.current_font = None  # current font
```
Comment (Lucas-C) [inline fpdf/fpdf.py:311]:
```suggestion
        self.current_font = None  # current font, None or an instance of TTFFont
```

**Comment by Lucas-C** [comment]
This is a really code refactoring!

Thank you @andersonhc 👍 

I just had a couple of minor comments, but this PR is excellent.

I'm eager to merge it 😄

**Comment by andersonhc** [comment]
> This is a really code refactoring!
> 
> Thank you @andersonhc 👍
> 
> I just had a couple of minor comments, but this PR is excellent.
> 
> I'm eager to merge it 😄

Thank you!

I'll start another PR soon migrating some functions to the font classes. It should be the last prep PR before text shaping.

**Comment by Lucas-C** [comment]
Merged!

**Pull request merged**